src/transformers/data/processors/glue.py

```diff
@@ -81,27 +81,42 @@ def _tf_glue_convert_examples_to_features(
 
         def gen():
             for ex in features:
-                yield (
+                if ex.token_type_ids is None:
+                    yield (
+                        {"input_ids": ex.input_ids, "attention_mask": ex.attention_mask},
+                        ex.label,
+                    )
+                else:
+                    yield (
+                        {
+                            "input_ids": ex.input_ids,
+                            "attention_mask": ex.attention_mask,
+                            "token_type_ids": ex.token_type_ids,
+                        },
+                        ex.label,
+                    )
```

Comment:
Those if/else statements are slightly verbose, maybe you should just keep the whole `ex` (minus the label)

Reply:
I would love to reduce this statement, but I'm not sure to follow what you meant sorry. Can you develop a bit please.

Reply:
something like this (untested):
```python
d = ex.clone()
label = d.pop("label")
yield (d, label)
```

Reply:
Nice! I will try that.